server/routerlicious/packages/routerlicious-base/src/alfred/runner.ts

```diff
@@ -112,6 +112,34 @@ export class AlfredRunner implements IRunner {
 
 		httpServer.on("error", (error) => this.onError(error));
 		httpServer.on("listening", () => this.onListening());
+		httpServer.on("upgrade", (req, socket, initialMsgBuffer) => {
+			const metric = Lumberjack.newLumberMetric("WebSocket Connections", {
+				origin: "upgrade",
+				connections: socket.server._connections,
+			});
+			metric.success("WebSockets: connection upgraded");
+			socket.on("close", (hadError: boolean) => {
+				const closeMetric = Lumberjack.newLumberMetric("WebSocket Connections", {
+					origin: "close",
+					connections: socket.server._connections,
+					hadError: hadError.toString(),
+				});
+				closeMetric.success(
+					"WebSockets: connection closed",
+					hadError ? LogLevel.Error : LogLevel.Info,
+				);
+			});
+			socket.on("error", (error) => {
+				const errorMetric = Lumberjack.newLumberMetric("WebSocket Connections", {
+					origin: "error",
+					connections: socket.server._connections,
+					bytesRead: socket.bytesRead,
+					bytesWritten: socket.bytesWritten,
+					error: error.toString(),
+				});
+				errorMetric.success("WebSockets: connection error", LogLevel.Error);
+			});
+		});
 
 		if (cluster.isPrimary && this.server.webSocketServer === null) {
 			// Listen on provided port, on all network interfaces.
```

Comment:
Can't comment because it's too far from changes, but this PR should be also adding an Alfred config to disable Alfred socket.io, right?

Reply:
In this PR no. As discussed in the design review, we will keep Alfred capable of handling web sockets connections if needed. A later PR will remove Alfred websockets

Reply:
To control traffic to nexus we will use an already existing configuration the deltaStreamURL (that's why it does not show up in this PR). We are not changing the default yet to Nexus. Currently it points to Alfred

Reply:
I'm not saying _remove_ it, I'm saying wrap it in a config which disables it. All good though. Probably doesn't hurt too much in terms of wasted resources if we don't send traffic to it.